src/core/model/OverlayBase.h

```diff
@@ -0,0 +1,17 @@
+/*
+ * Xournal++
+ *
+ *
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+class OverlayBase {
+public:
+    virtual ~OverlayBase() = default;
+};
```

Comment:
What's the use case of an empty base class, without virtual functions to call?

Reply:
Wait no! There was a use case: see `XojPageView::deleteViewOf(OverlayBase*)`. I just needed a common pointer type for everything that will be an overlay (for now, only InputHandler, but more should come).

Reply:
Then OverlayBase should named ViewBase, and it should require all view functions defined from Repaintable.

Reply:
No, it's not a view. It's a base for some controllers: those controllers which can have one or more views. This includes all the tool controllers (by that I mean things in control/tools), but also SearchControl for instance (to highlight the search results).

It is obviously not clear or we wouldn't be having this discussion, so something needs to be done to clarify the situation. Not sure what yet.